src/DynamicRupture/FrictionLaws/FastVelocityWeakeningLaw.cpp

```diff
@@ -0,0 +1,186 @@
+                                                                        unsigned int timeIndex,
+                                                                        unsigned int ltsFace) {
+  real localTemperature = 0.0;
+  real localPressure = 0.0;
+
+  real tauV = faultStrength[pointIndex] *
+              slipRateMagnitude[ltsFace][pointIndex]; //! fault strenght*slip rate
+  real lambdaPrime = drParameters.tP_lambda * drParameters.alpha_th /
+                     (alphaHy[ltsFace][pointIndex] - drParameters.alpha_th);
+
+  real tmp[TP_grid_nz];
+  real omegaTheta[TP_grid_nz]{};
+  real omegaSigma[TP_grid_nz]{};
+  real thetaCurrent[TP_grid_nz]{};
+  real sigmaCurrent[TP_grid_nz]{};
+  for (unsigned int iTP_grid_nz = 0; iTP_grid_nz < TP_grid_nz; iTP_grid_nz++) {
+    //! Gaussian shear zone in spectral domain, normalized by w
+    tmp[iTP_grid_nz] =
+        std::pow(TP_grid[iTP_grid_nz] / TP_halfWidthShearZone[ltsFace][pointIndex], 2);
+    //! 1. Calculate diffusion of the field at previous timestep
+
+    //! temperature
+    thetaCurrent[iTP_grid_nz] =
+        Theta_tmp[iTP_grid_nz] * exp(-drParameters.alpha_th * deltaT[timeIndex] * tmp[iTP_grid_nz]);
+    //! pore pressure + lambda'*temp
+    sigmaCurrent[iTP_grid_nz] = Sigma_tmp[iTP_grid_nz] * exp(-alphaHy[ltsFace][pointIndex] *
+                                                             deltaT[timeIndex] * tmp[iTP_grid_nz]);
+
+    //! 2. Add current contribution and get new temperature
+    omegaTheta[iTP_grid_nz] =
+        heatSource(tmp[iTP_grid_nz], drParameters.alpha_th, iTP_grid_nz, timeIndex);
+    Theta_tmp[iTP_grid_nz] =
+        thetaCurrent[iTP_grid_nz] + (tauV / drParameters.rho_c) * omegaTheta[iTP_grid_nz];
+    omegaSigma[iTP_grid_nz] =
+        heatSource(tmp[iTP_grid_nz], alphaHy[ltsFace][pointIndex], iTP_grid_nz, timeIndex);
+    Sigma_tmp[iTP_grid_nz] =
+        sigmaCurrent[iTP_grid_nz] + ((drParameters.tP_lambda + lambdaPrime) * tauV) /
+                                        (drParameters.rho_c) * omegaSigma[iTP_grid_nz];
+
+    //! 3. Recover temperature and pressure using inverse Fourier
+    //! transformation with the calculated fourier coefficients
+
+    //! new contribution
+    localTemperature += (TP_DFinv[iTP_grid_nz] / TP_halfWidthShearZone[ltsFace][pointIndex]) *
+                        Theta_tmp[iTP_grid_nz];
+    localPressure += (TP_DFinv[iTP_grid_nz] / TP_halfWidthShearZone[ltsFace][pointIndex]) *
+                     Sigma_tmp[iTP_grid_nz];
+  }
+  // Update pore pressure change (sigma = pore pressure + lambda'*temp)
+  // In the BIEM code (Lapusta) they use T without initial value
+  localPressure = localPressure - lambdaPrime * localTemperature;
+
+  // Temp and pore pressure change at single GP on the fault + initial values
+  temperature[ltsFace][pointIndex] = localTemperature + drParameters.initialTemperature;
+  pressure[ltsFace][pointIndex] = -localPressure + drParameters.iniPressure;
```

Comment:
Maybe we need to be consistent with `initial` and `ini` and stick to one style?

Reply:
It's initial everywhere now